### PR TITLE
Support SELF typing for OracleContainer

### LINKED ISSUE #1423: OracleContainer extends JdbcDatabaseContainer without generic type
It means I need explicit cast when using methods from super type, eg:
```java
@ClassRule
public static OracleContainer db = (OracleContainer) new OracleContainer("foo")
        .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("db container logger")));
```
Following screams compilation errors, because `#withLogConsumer` returns `GenericContainer`:
```java
@ClassRule
public static OracleContainer db = new OracleContainer("foo")
        .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("db container logger")));
```

### PR DESCRIPTION
**Pull request by mumukiller**
Added waitUntilContainerStarted method
Adjusted OracleContainer class declaration

Fixes #1423

### CODE REVIEW

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /**
  * @author gusohal
  */
-public class OracleContainer extends JdbcDatabaseContainer {
+public class OracleContainer<SELF extends OracleContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
```
Comment (bsideup) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: It should be:
```suggestion
public class OracleContainer extends JdbcDatabaseContainer<OracleContainer> {
```

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java [hunk 2]
```diff
@@ -108,4 +108,9 @@ public Integer getWebPort() {
     public String getTestQueryString() {
         return "SELECT 1 FROM DUAL";
     }
+
+    @Override
+    protected void waitUntilContainerStarted() {
```
Comment (bsideup) [inline modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: why do you need to override it?

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java (hunk 1 above)
Comment (mumukiller) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: So PostgreSQLContainer and MySQLContainer have also legacy class declaration, is it correct?

Comment (bsideup) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: yes, I think most of them

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java (hunk 2 above)
Comment (mumukiller) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: I checked PostgreSQLContainer and it overrides waitUntilContainerStarted method but after your comment i also checked MySQLContainer which does not. I suppose that WaitStrategy for most JdbcDatabaseContainer containers is the same.

Comment (bsideup) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: PostgreSQLContainer uses log-based wait strategy, while others inherit the JDBC one, hence the override

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java (hunk 1 above)
Comment (mumukiller) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: Fixed

File: modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java (hunk 2 above)
Comment (mumukiller) [reply modules/oracle-xe/src/main/java/org/testcontainers/containers/OracleContainer.java]: fixed

**Comment by rnorth** [comment]
Thank you @mumukiller!

**Comment by rnorth** [comment]
Released in [1.11.3](https://github.com/testcontainers/testcontainers-java/releases/tag/1.11.3). Thanks for the contribution!

**Pull request merged**